src/lib/pipeline/FetchUnit.cc

```diff
@@ -53,6 +53,12 @@ void FetchUnit::tick() {
       // Set prediction to recorded value during loop buffer filling
       if (macroOp[0]->isBranch()) {
         macroOp[0]->setBranchPrediction(loopBuffer_.front().prediction);
+        [[maybe_unused]] BranchPrediction pred = branchPredictor_.predict(
```

Comment:
+1

Reply:
Could you be more specific about _why_ the branch needs to be "logged" with the BP. Justify why the me in 2 years shouldn't remove that line. Refer to the specific types of structures that need to be updated e.g. history tracking via global history and FTQ etc.

The assert is now broken because `pred` is not defined. Having the maybe unused variable was fine, it just needed some explanation.

Reply:
Comment updated

Reply:
The updated comment is better however it is now out of date since you removed the assert. Having extra asserts causes no harm IMO and I think this one was important so I think you should put that back.

Reply:
I've updated the comment.  The assert is not important which is why i removed it.  Having thought about it more, it does not matter if they are the same or not.  The BP still updates itself on the basis of what it would have predicted, and what the branch ended up being.  The prediction used by the pipeline does not feed back into the BP on update.

Reply:
Ah ok, that makes sense. In this case you can now remove the pred variable and have a shorter comment as you did some commits ago

Reply:
Maybe state the we are "ignoring the result"